src/Cli/dotnet/ToolPackage/ToolPackageDownloader.cs

```diff
@@ -107,11 +107,12 @@ public IToolPackage InstallPackage(PackageLocation packageLocation, PackageId pa
                     }
                     NuGetVersion packageVersion = nugetPackageDownloader.GetBestPackageVersionAsync(packageId, versionRange, packageSourceLocation).GetAwaiter().GetResult();
 
-                    rollbackDirectory = isGlobalTool ? toolDownloadDir.Value: Path.Combine(toolDownloadDir.Value, packageId.ToString(), packageVersion.ToString());
+                    rollbackDirectory = isGlobalTool ? toolDownloadDir.Value: new VersionFolderPathResolver(toolDownloadDir.Value).GetInstallPath(packageId.ToString(), packageVersion);
 
                     if (isGlobalTool)
                     {
-                        NuGetv3LocalRepository nugetPackageRootDirectory = new(Path.Combine(_toolPackageStore.GetRootPackageDirectory(packageId).ToString().Trim('"'), packageVersion.ToString()));
+                        // TODO: NuGetv3LocalRepository already has a VersionFolderPathResolver member, so this can probably be simplified
```

Comment:
If you don't plan on including this in this PR (which you dont have to) I would suggest you remove the TODO and add a remark in the issue mentioned in an earlier comment.